apps/condo/domains/notification/schema/Device.test.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * Generated by `createschema notification.Device 'deviceId:Text; pushToken?:Text; pushTransport?:Select:firebase,apple,huawei; owner?:Relationship:User:SET_NULL; meta?:Json'`
+ */
+const faker = require('faker')
+const sample = require('lodash/sample')
+
+const { makeLoggedInAdminClient, makeClient, makeLoggedInClient, UUID_RE, DATETIME_RE } = require('@core/keystone/test.utils')
+
+const {
+    expectToThrowValidationFailureError,
+    expectToThrowAuthenticationErrorToObjects,
+    expectToThrowAccessDeniedErrorToObj,
+    expectToThrowMutationError,
+    expectToThrowUserInputError,
+} = require('@condo/domains/common/utils/testSchema')
+
+const { Device, createTestDevice, updateTestDevice } = require('@condo/domains/notification/utils/testSchema')
+
+const { PUSH_TRANSPORT_TYPES } = require('../constants/constants')
+
+const DUPLICATE_CONSTRAINT_VIOLATION_ERROR_MESSAGE = 'duplicate key value violates unique constraint'
+
+describe('Device', () => {
```

Comment:
Could you please reorganise test cases in a way, that they will be grouped by operations and for each operation a roles ability is examined.
Example can be found in `apps/condo/domains/resident/schema/Resident.test.js`.

A statement `it('can be updated by admin')` reads more naturally, than `it('admin: direct update access allowed')`.
A function `it` is created exactly for this reason, — to make the code feel like a natural language.

Reply:
fixed